tests/test_main.py

```diff
@@ -1,3 +1,76 @@
 from app.main import validate_email
 
-# write your tests here
+import random
+import string
+
+import pytest
+
+
+def email_generator():
```

Comment:
This email_generator may contain a lot of problems.
It is not tested - and is used in tests. 
I cannot say, that this code is good.
You'd better write simpler tests and check all classes of equivalence, rather than such an approach.

Reply:
I agree, I’ll rewrite the tests

Reply:
I've fixed, and found 2 bugs.